docs/api/serviceregistry/device-discovery-generic-http.md

```diff
@@ -1,25 +1,205 @@
+
+```
+{
+   "name":"thermometer2",
+   "metadata":{
+      "scales":[
+         "Kelvin",
+         "Celsius"
+      ],
+      "max-temperature":{
+         "Kelvin":310,
+         "Celsius":40
+      },
+      "min-temperature":{
+         "Kelvin":260,
+         "Celsius":-10
+      }
+   },
+   "addresses":[
+      {
+         "type":"MAC",
+         "address":"81:ef:1a:44:7a:f5"
+      }
+   ],
+   "createdAt":"2024-11-04T01:53:02Z",
+   "updatedAt":"2024-11-04T01:53:02Z"
+}
+```
+The **error codes** are, `400 Bad Request` if request is malformed, `403 Forbidden` if requester au-
+thentication was unsuccessful, `401 Unauthorized` if the authenticated requester has no permission and
+`500 Internal Server Error` if an unexpected error happens. The error response also contains an
+[ErrorResponse](../data-models/error-response.md) JSON encoded body.
+
+```
+{
+    "errorMessage": "Device name is missing.",
+    "errorCode": 400,
+    "exceptionType: "INVALID_PARAMETER",
+    "origin": "POST /serviceregistry/device-discovery/register"
+}
+```
 
 ### lookup
 
-TODO
+The service operation **request** requires an authorization bearer header and may optionally include a [DeviceLookupRequest](../data-models/device-lookup-request.md) JSON encoded body.
+
+```
+POST /serviceregistry/device-registry/lookup HTTP/1.1
+Authorization: Bearer <authorization-info>
+
+{
+   "deviceNames":[
+      "thermometer2"
+   ],
+   "addresses":[
+      "81:ef:1a:44:7a:f5"
+   ],
+   "addressType":"MAC",
+   "metadataRequirementList":[
+      {
+         "max-temperature.Celsius":{
+            "op":"LESS_THAN",
+            "value":50
+         }
+      }
+   ]
+}
+```
+
+The service operation **responses** with the status code `200 Ok` if called successfully and with a [DeviceLookupResponse](../data-models/device-lookup-response.md) JSON encoded body.
+
+```
+{
+   "entries":[
+      {
+         "name":"thermometer2",
+         "metadata":{
+            "scales":[
+               "Kelvin",
+               "Celsius"
+            ],
+            "max-temperature":{
+               "Kelvin":310,
+               "Celsius":40
+            },
+            "min-temperature":{
+               "Kelvin":260,
+               "Celsius":-10
+            }
+         },
+         "addresses":[
+            {
+               "type":"MAC",
+               "address":"81:ef:1a:44:7a:f5"
+            }
+         ],
+         "createdAt":"2024-11-04T01:53:02Z",
+         "updatedAt":"2024-11-04T01:53:02Z"
+      }
+   ],
+   "count":1
+}
+```
+
+The error codes are, `400 Bad Request` if request is malformed, `403 Forbidden` if requester authentication was unsuccessful, `401 Unauthorized` if the authenticated requester has no permission and `500 Internal Server` Error if an unexpected error happens. The error response also contains an [ErrorResponse](../data-models/error-response.md) JSON encoded body.
```

Comment:
It is confusing, but 401 is for authentication and 403 is for authorization problems